dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java

```diff
@@ -386,7 +386,9 @@ private void refreshInvokers(BitList<Invoker<T>> targetInvokers, Collection<Invo
 
     @Override
     public void addDisabledInvoker(Invoker<T> invoker) {
-        MetricsEventBus.publish(DirectorSupport.disable(applicationModel));
+        if (invoker.getInterface() != null) {
+            MetricsEventBus.publish(DirectorSupport.disable(applicationModel, invoker.getInterface().getName()));
```

Comment:
Can we replace with the exact invokers count in the collector after modify

Reply:
ok